cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/trigger/AbstractCompositeTrigger.java

```diff
@@ -85,4 +88,14 @@ private void initializeUnitTriggers() {
       }
     }
   }
+
+  public List<TriggerInfo> getUnitTriggerInfos(TriggerInfoContext context) {
```

Comment:
yes, a composite trigger can contain other composite triggers, however, this method will only return the triggers which are not composite triggers, since composite triggers get trigger keys, trigger info and etc. from the internal unit triggers (i.e. non-composite triggers). I think LinkedHashMultimap will be better since it avoids duplicates in values?